enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,532 @@
+  but it is not entirely connected with the operator itself and is more CNI plugin specific.
+
+For the [phase 1](#implementation-phases) of this project, there seems to be no particular action or API knobs needed regarding
+the addition of the "External" platform type. Just ensure that CNO is non-reactive to the "External" platform type
+and behaves the same as in the "None" platform case.
+In the future, we possibly might want to make CNO more tolerant of partner CNCC implementations
+and design a way for configuring platform-specific CNI behaviour.
+
+### Workflow Description
+
+During [phase 1](#implementation-phases) proposed changes intended to have no effect and the "External"
+platform type should be handled the same as the "None", so no specific user interaction is expected.
+
+### API Extensions
+
+A new optional constant of `PlatformType` type will be added to the "openshift/api".
+
+```go
+const (
+    ...
+    // ExternalPlatformType represent generic infrastructure provider. Provider-specific components should be supplemented separately.
+    ExternalPlatformType PlatformType = "External"
+    ...
+)
+```
+
+Additionally, the respective external platform spec and status should be added to the infrastructure resource.
+
+```go
+type CloudControllerManagerMode string
+
+const (
+    // Cloud Controller Manager is enabled and expected to be supplied.
+    // Signaling MCO to set `--cloud-provider=external` flag to the kubelets.
+    CloudControllerManagerEnabled CloudControllerManagerMode = "Enabled"
+    // No Cloud Controller Manager is expected to be supplied.
+    // Signaling MCO not to set `--cloud-provider` flag to the kubelets.
+    CloudControllerManagerDisabled CloudControllerManagerMode = "Disabled"
+)
+
+type CloudControllerManagerSettings struct {
+    // Mode determines whether or not an external Cloud Controller Manager is expected to 
+    // be presented in the cluster.
+    // For engaging an external Cloud Controller Manager, certain flags are expected to be set to the kubelets.
+    // https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+    //
+    // When enabled, the respective operator (machine config operator) should set `--cloud-provider=external` flag to the kubelet.
+    // When omitted or disabled, no `cloud-provider` flag should be set.
+    // +kubebuilder:validation:Enum=Enabled;Disabled
+    // +optional
+    Mode CloudControllerManagerMode `json:"mode,omitempty"`
+}
+
+
+// ExternalPlatformSpec holds the desired state for the generic External infrastructure provider.
+type ExternalPlatformSpec struct{
+    // ProviderName holds the arbitrary string represented cloud provider name, expected to be set at the installation time.
+    // Intended to serve only for informational purposes and not expected to be used for decision-making.
+    // +kubebuilder:default:="Unknown"
+    // +optional
+    ProviderName string `json:"providerName,omitempty"`
```

Comment:
Are we intending to have this as immutable once set to something other than `Unknown`?

Reply:
Unset would be the empty string right? Unknown seems like a reasonable default to me for now

Reply:
I would like to leave "Unknown" as default.

>Are we intending to have this as immutable once set to something other than Unknown?
Yup, that feels right to me.